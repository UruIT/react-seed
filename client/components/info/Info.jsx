import React from 'react';
import style from './Info.scss';
import Modal from '../modal/Modal';

export default class Info extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			isModalOpen: false,
			enableCloseOnBackgroundClick: true,
			contentWidth: 100,
			contentHeight: 100
		};
		this.handleChange = this.handleChange.bind(this);
	}

	render() {
		return (
			<div className={style.info}>
				<label>
					Close modal on background clicking:
					<input
						type="checkbox"
						name="enableCloseOnBackgroundClick"
						checked={this.state.enableCloseOnBackgroundClick}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Conent WIDTH (in px):
					<input
						type="text"
						name="contentWidth"
						placeholder="width (px)"
						value={this.state.contentWidth}
						onChange={this.handleChange}
					/>
				</label>
				<label>
					Conent HEIGHT (in px):
					<input
						type="text"
						name="contentHeight"
						placeholder="height (px)"
						value={this.state.contentHeight}
						onChange={this.handleChange}
					/>
				</label>
				<input type="submit" onClick={() => this.openModal()} value="OPEN DIALOG" />
				<Modal
					open={this.state.isModalOpen}
					onClose={() => this.closeModal()}
					closeClickingOutside={this.state.enableCloseOnBackgroundClick}
				>
					<div
						className={style.content}
						style={{ width: this.state.contentWidth, height: this.state.contentHeight }}
					>
						{' '}CONTENT BOX <br /> Size: {this.state.contentWidth} px X {this.state.contentHeight} px{' '}
					</div>
					<button onClick={() => this.closeModal()}>Close</button>
				</Modal>
			</div>
		);
	}

	handleChange(e) {
		let value;
		if (e.target.type === 'checkbox') {
			value = e.target.checked;
		} else {
			value = e.target.value;
		}
		this.setState({ [e.target.name]: value });
	}

	openModal() {
		this.setState({ isModalOpen: true });
	}

	closeModal() {
		this.setState({ isModalOpen: false });
	}
}
