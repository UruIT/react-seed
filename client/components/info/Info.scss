.info {
	text-align: center;

	input[type="submit"] {
		display: block;
		padding: 10px 20px;
		border-radius: 5px;
		margin: 10px auto;
	}

	label {
		display: block;
		font-size: 20px;
		padding: 10px;
		margin: 10px;

		input {
			padding: 10px;
			margin: 0 10px;
			border-radius: 5px;
		}

		input[type="checkbox"] {
			width: 25px;
			height: 25px;
		}
	}

	.content {
		border: solid 1px #0A0817;
		text-align: center;
		background-color: #5882FA;
	}
}
