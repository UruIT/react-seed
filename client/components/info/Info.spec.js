import React from 'react';
import Info from './Info';
import renderer from 'react-test-renderer';

test('renders Info', () => {
	const tree = renderer.create(<Info />).toJSON();

	expect(tree).toMatchSnapshot();
});
