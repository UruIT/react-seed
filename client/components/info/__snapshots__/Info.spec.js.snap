// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Info 1`] = `
<div
  className="info"
>
  <label>
    Close modal on background clicking:
    <input
      checked={true}
      name="enableCloseOnBackgroundClick"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <label>
    Conent WIDTH (in px):
    <input
      name="contentWidth"
      onChange={[Function]}
      placeholder="width (px)"
      type="text"
      value={350}
    />
  </label>
  <label>
    Conent HEIGHT (in px):
    <input
      name="contentHeight"
      onChange={[Function]}
      placeholder="height (px)"
      type="text"
      value={400}
    />
  </label>
  <input
    onClick={[Function]}
    type="submit"
    value="OPEN DIALOG"
  />
</div>
`;
